src/main/java/org/jenkinsci/plugins/buildnamesetter/BuildNameSetter.java

```diff
@@ -26,50 +28,106 @@
-                return true;
+                try {
+                    setDisplayName(build, listener);
+	            return true;                   
+                } catch (IllegalArgumentException e){
+                    listener.getLogger().println(e.getMessage());   
+                }
+                return false;
             }
-        };
+         };
     }
-
-    private void setDisplayName(AbstractBuild build, BuildListener listener) throws IOException, InterruptedException {
-        try {
-            build.setDisplayName(TokenMacro.expandAll(build, listener, template));
-        } catch (MacroEvaluationException e) {
+    private void setDisplayName(AbstractBuild build, BuildListener listener) throws IOException, InterruptedException, IllegalArgumentException {
+    	try{
+            switch(scenario) {
+            case 0:         
+                build.setDisplayName(TokenMacro.expandAll(build, listener, matrixTemplate));
+                return;
+            case 1:
+                if(started) {	
+                    build.setDisplayName(TokenMacro.expandAll(build, listener, matrixTemplate));
+                }else {
+                    build.setDisplayName(TokenMacro.expandAll(build, listener, template));
+                }
+                return;
+            case 2:
+                build.setDisplayName(TokenMacro.expandAll(build, listener, template));
+                return;
+            case -1:
+                listener.getLogger().println("No valid values set for build names!");
+            }
+    	} catch (MacroEvaluationException e) {
             listener.getLogger().println(e.getMessage());
-        }
+    	}  
     }
 
     public MatrixAggregator createAggregator(MatrixBuild build, Launcher launcher, BuildListener listener) {
         return new MatrixAggregator(build,launcher,listener) {
             @Override
             public boolean startBuild() throws InterruptedException, IOException {
-                setDisplayName(build,listener);
-                return super.startBuild();
+                try {
+                    setDisplayName(build,listener);
+                    started = true;
+	            return super.startBuild();
+                } catch (IllegalArgumentException e){
+                    listener.getLogger().println(e.getMessage());   
+                }
+                return false;
             }
 
             @Override
             public boolean endBuild() throws InterruptedException, IOException {
-                setDisplayName(build,listener);
-                return super.endBuild();
+                try{
+                    setDisplayName(build,listener);
+                    return super.endBuild();
+                } catch (IllegalArgumentException e){
+                    listener.getLogger().println(e.getMessage());   
+                }
+                return false;
             }
         };
     }
+   
 
-    @Extension
+    @Extension(optional = true)
```

Comment:
Okay. I got that tag from the OSS wiki [on plugin dependencies](https://wiki.jenkins-ci.org/display/JENKINS/Dependencies+among+plugins): 

> You can mark your extension as an optional extension by changing the @Extension annotation to @Extension(optional = true) and Jenkins don't log any class loading errors when reading it.

I'll look through the options [on this page](https://wiki.jenkins-ci.org/display/JENKINS/Tips+for+optional+dependencies) instead.